backend/app/Models/User.php

```diff
@@ -37,4 +37,36 @@ public function getToken()
     {
         return auth()->login($this);
     }
+
+    /**
+     * Return boolean, whether or not user is following user_id.
+     *
+     * @return bool
+     */
+    public function isFollowing($user_id)
+    {
+        return Follow::where([
+            ['user_id', '=', $user_id],
+            ['follower_id', '=', $this->id],
+        ])->exists();
+    }
+
+    /**
+     * Follow user_id.
+     *
+     * @return bool True - if user is successfully following user_id, False - if user is already following user_id
+     */
+    public function follow($user_id)
+    {
+        if (!$this->isFollowing($user_id)) {
+            $follow = new Follow();
+            $follow->user_id = $user_id;
+            $follow->follower_id = $this->id;
```

Comment:
This can be done with laravel's `attach()` method: https://laravel.com/docs/5.6/eloquent-relationships#inserting-and-updating-related-models (see Many To Many Relationships - Attaching / Detaching)

```
$user = App\User::find(1);
$user->roles()->attach($roleId);
```